types/index.d.ts

```diff
@@ -656,7 +650,7 @@ declare module 'mongoose' {
    * It creates dot notation for arrays similar to mongodb. For example { a: { c: { b: number}[] }[] } => 'a.c.b': number, 'a.c': { b: number }[]
    */
   type DotKeys<DocType> = {
-    [key in DotNestedKeys<ExtractNestedArrayElement<DocType>, DotnotationMaximumDepth>]?: FindDottedPathType<NestedRequired<DocType>, key>;
+    [key in DotNestedKeys<ExtractNestedArrayElement<DocType>, DotnotationMaximumDepth>]?: any;
```

Comment:
With `any` we do not support typing in dot notations anymore.

Reply:
I'm open to better solutions, if you can figure something out that fixes points 1-3 in OP that would be amazing. But right now, with tests failing I can't deploy 8.1.

Reply:
The implementation of `3` looks good.
The test are failing after this commit on 8.1: https://github.com/Automattic/mongoose/commit/7299980697c4ef1c0cfeb1128dda6474ee9a177b

I look into it.

Reply:
Thanks :+1: